2018/002-update-channels.md

```diff
@@ -0,0 +1,145 @@
+upgrade from 1.10.x to 1.11.x, default pod security policy introduction).
+
+Right now there is no visibility of what is inside an update and furthermore
+there is no possible way around intrusive updates by having the ability to only
+apply critical bug fixes and security updates. An admin needs to apply all updates
+that come through a single channel.
+
+In order to separate intrusive updates in SUSE CaaS Platform, we need different
+update channels. This allows the separation of possibly disruptive updates
+into their own channels, and lets the admin continue applying bugfixes and security
+updates without the need to apply intrusive, new features. The admin can also see
+what is inside an update to be able to decide when to apply it.
+
+The channels will also act as a barrier to avoid that a user runs k8s 1.8 and
+upgrades right away to 1.11 in one step.
+
+### Proposed change
+
+The admin will have the ability of choosing when to switch to the next channel.
+
+In case of an airgapped system, a link to the release notes of the update channel
+is displayed. A system with access to the web can fetch the release notes and
+display them in the UI directly. In both scenarios, an admin can evaluate the
+upcoming changes beforehand.
+
+The UI of velum needs to be adapted to enable a switch to a new channel.
+
+## Detailed RFC
+
+We can separate the updates into different channels. There are 2 types of channels:
+
+* update channel: normal maintenance updates, which are safe and non-intrusive
+* new version channel: version updates which can contain intrusive updates
+
+This RFC proposes:
+
+* a front-end change in the velum UI to enable the admin to switch to a new version
+channel
+* a back-end change in salt to perform the channel switch on all cluster nodes
+
+### Proposed change (Detailed)
+
+#### Tools/Services
+
+We need a set of tools/services to indicate, whenever a maintenance update or a new version
+update is ready to be installed. They should run on each worker node to be more independent
+and decoupled from velum. If velum would be the initiator of the update check, it might lead
+to inconsistent and long execution runs in larger clusters.
+
+* "check for updates" service on each worker
+  - reports availability of maintenance updates to velum (via grains?)
+  - once per day e.g.
+  - decoupled from velum, runs independently, just reports
+* "new version" checker service on each worker
+  - check for new versions (migration targets)
+    - query RMT/SMT/SCC API if there is a new compatible version available for this CaaSP version (SUSEConnect library)
+    - report back to velum (via grains?)
+  - get the link to the release notes
+  - once per week or longer timeframe
+  - decoupled from velum, runs independently, just reports
+
+#### User Interface
+
+The user interface of velum needs to be adapted minimally.
+
+* [indirect] display of update content through [a link to the] release notes
+* display of update history via velum
```

Comment:
What kind of information is going to be kept?

Reply:
i'd say the same information that gets displayed when a new channel is there? (e.g. package list, release notes, or link to release notes in case of offline setup)

so this will then become history plus maybe a timestamp when it happened?

Reply:
I see. I would not make that part of the first iteration of the feature though